forest/jasmine/traj2stats.py

```diff
@@ -58,6 +63,90 @@ def transform_point_to_circle(lat: float, lon: float, radius: int
     return transform(aeqd_to_wgs84, buffer)
 
 
+def get_nearby_locations(traj: np.ndarray
+                         ) -> Tuple[dict, dict, dict]:
+    """This function returns a dictionary of nearby locations,
+    a dictionary of nearby locations' names, and a dictionary of
+    nearby locations' coordinates.
+
+    Args:
+        traj: numpy array, trajectory
+    Returns:
+        nearby_locations: dictionary, nearby locations
+        nearby_locations_name: dictionary, nearby locations' names
+        nearby_locations_coord: dictionary, nearby locations' coordinates
+    Raises:
+        RuntimeError: if the query to Overpass API fails
+    """
+
+    pause_vec = traj[traj[:, 0] == 2]
+    lat: List[float] = []
+    lon: List[float] = []
+    for row in pause_vec:
+        if len(lat) == 0:
+            lat.append(row[1])
+            lon.append(row[2])
+        # only add locations that are far away (> 1km)
+        elif (
+            np.min(
+                    [
+                        great_circle_dist(*row[1:3], lat[i], lon[i])
+                        for i in range(len(lat))
+                    ]
+                )
+            > 1000
+        ):
+            lat.append(row[1])
+            lon.append(row[2])
```

Comment:
I think this conditional can be much easier to read by doing something like:
```
distance = np.min([...])
if len(lat) == 0 or distance > 1000
    ...
```
It'd be also great to explain what does `len(lat) == 0` mean